### ISSUE TITLE
Provide official Docker image on Docker hub

### ISSUE THREAD

**Issue by alexanderadam** [geminabox/geminabox#326]
It seems that there is no _official_ image for geminabox on Docker Hub.
I can find only [various inofficial ones](https://hub.docker.com/search?q=geminabox&type=image).

Having official images would probably help everyone looking for a proper geminabox installation that won't be outdated soonish.

**Comment by nacengineer** [comment]
I get the gist behind this but geminabox is already only a gem that runs via rack. It's pretty simple to install as is and this seems out of scope.

**Comment by alexanderadam** [comment]
> geminabox is already only a gem that runs via rack. It's pretty simple to install as is and this seems out of scope.

Well, I thought this would be obvious but:

container images are not just about a simple installation. It's also about

- having a secure, isolated environment where an app runs (so a bug in geminabox won't necessary lead to exposing all other applications that are running on that machine)
- fast and simple rollbacks
- ensured compatibility for whatever underlying OS or packages you have
- usually faster configuration and integration with things like træfik (so you don't even have to 'configure' a load balancer or domain rules)

This is the reason why you will even find Docker images for tools that have a single binary and no other dependencies on Docker hub.

**Comment by nacengineer** [comment]
Don't get me wrong I love the container idea *but* all of those things are on the implementation side. So it adds unnecessary scope creep for this repository. The deploy scope doesn't/shouldn't really apply to this repo which is for the gem itself. 

It's essentially the same argument that was had on here for tearing out the capistrano bits from this repo, the deployment is separate from the gem code itself.

**Comment by alexanderadam** [comment]
> So it adds unnecessary scope creep for this repository. The deploy scope doesn't/shouldn't really apply to this repo which is for the gem itself.

I have the feeling that this discussion is drifting apart. :wink:

However, usually all dependencies should be documented within the same repository.
This includes the `Dockerfile` (and related) of course because if you don't have all dependencies in the Dockerfile, your application will most probably _not_ work, right? :wink: 

Having separate repositories would lead to higher maintenance or inconsistent versioning. This is why you will usually find Docker configuration in the same repository (see for example [GitLab](https://gitlab.com/gitlab-org/gitlab-foss/tree/master#docker-compose.yml), [Thredded](https://github.com/thredded/thredded#3254677a7917c6c01f55212f86c57fbf-98378af556ca833ffefdf3bbee8ac453ca9ac613), [OpenProject](https://github.com/opf/openproject) or [Black Candy](https://github.com/aidewoode/black_candy) to name a few).

Of course no one _have to_ do it this way but you will see that it is pretty common.

**Comment by nacengineer** [comment]
@wrein it's been 4 years. Today I'd probably walk back most of what I said as I think adding a Dockerfile to a repo has mostly become SOP in the intervening years.

**Comment by wrein** [comment]
Indeed, 4 years feels like having past a WWIII, sort of, let's ask @tnir to reopen this issue...

If that's the polite way to do things, I'm not sure.

**Comment by alexanderadam** [comment]
I'm glad you changed your mind.
I just reopened this issue although users looking for a gem server might also want to [checkout the GitLab documentation](https://docs.gitlab.com/ee/user/packages/rubygems_registry/) in case they're already running GitLab.

Because from GitLab 13.10 on there's RubyGems support available as a preview feature.
[Some things are still missing though](https://gitlab.com/groups/gitlab-org/-/epics/3200).

**Comment by wrein** [comment]
@alexanderadam, did you read my mind? I really appreciate your input because I'm on GitLab 14.9 and... it doesn't really works yet, for me. I passed last two nights to trying to setup my GitLab instance and ... nope, nix, niet, nein.  

> Because from GitLab 13.10 on there's RubyGems support available as a preview feature. [Some things are still missing though](https://gitlab.com/groups/gitlab-org/-/epics/3200).
I am planning an upgrade, but I am delaying it because I cannot afford to stop the server for the time being.
If anyone has positive feedback on gem support on the newer versions I will be encouraged to upgrade.

I went for geminabox precisely because I was discouraged by GitLab.
